src/Libplanet.Types/Assets/Currency.cs

```diff
@@ -49,201 +47,28 @@ namespace Libplanet.Types.Assets
     /// </remarks>
     /// <seealso cref="FungibleAssetValue"/>
     [Serializable]
-    public readonly struct Currency : IEquatable<Currency>, ISerializable
+    public sealed class Currency : IEquatable<Currency>, ISerializable
     {
-        /// <summary>
-        /// The ticker symbol, e.g., <c>&quot;USD&quot;</c>.
-        /// </summary>
-        [JsonInclude]
-        public readonly string Ticker;
-
-        /// <summary>
-        /// The number of digits to treat as <a
-        /// href="https://w.wiki/ZXv#Treatment_of_minor_currency_units_(the_%22exponent%22)">minor
-        /// units (i.e., exponent)</a>.
-        /// </summary>
-        [JsonInclude]
-        public readonly byte DecimalPlaces;
-
-        /// <summary>
-        /// The <see cref="Address"/>es who can mint the currency.
-        /// If this is <see langword="null"/> <em>anyone</em> can mint the currency.
-        /// </summary>
-        /// <remarks>
-        /// Unlike <see langword="null"/>, an empty set means <em>no one</em> can mint the currency.
-        /// </remarks>
-        /// <seealso cref="Libplanet.Action.State.IAccount.MintAsset"/>
-        [JsonInclude]
-        public readonly IImmutableSet<Address>? Minters;
-
-        /// <summary>
-        /// The deterministic hash derived from other fields.
-        /// </summary>
-        [JsonInclude]
-        public readonly HashDigest<SHA1> Hash;
-
-        /// <summary>
-        /// Whether the total supply of this instance of <see cref="Currency"/> is trackable.
-        /// </summary>
-        [JsonInclude]
-        public readonly bool TotalSupplyTrackable;
+        private static Codec _codec = new Codec();
```

Comment:
My IDE said that `Make field readonly[IDE0044]` I would appreciate it if you could make it readonly 😀